### PR TITLE
Add Pull Refresh to AloeStackView

### PR DESCRIPTION
**Pull request by chuongtrh**
Please help me review a PR
Thanks :)

### CODE REVIEW

File: Sources/AloeStackView/AloeStackView.swift [hunk 1]
```diff
@@ -20,506 +20,531 @@ import UIKit
  * power of Auto Layout.
  */
 open class AloeStackView: UIScrollView {
+    // MARK: Lifecycle
```
Comment (marlimox) [inline Sources/AloeStackView/AloeStackView.swift]: Please revert all of these style changes.

**Comment by marlimox** [comment]
Hi @chuongtrh, thank you for your contribution.

Could you please first revert all of the indentation and style changes in this PR? Thanks!

File: Sources/AloeStackView/AloeStackView.swift (hunk 1 above)
Comment (chuongtrh) [reply Sources/AloeStackView/AloeStackView.swift]: Yes, I just committed again.

**Comment by marlimox** [comment]
Thank you for your contribution @chuongtrh! My feeling on this change is that I think it would be better for call sites to hook up pull-to-refresh themselves, rather than modifying AloeStackView to provide this functionality.

Here are my thoughts on why:

Adding pull-to-refresh to AloeStackView is possible via the existing public API of UIScrollView. It doesn’t require a change to AloeStackView to work.

Pull-to-refresh isn’t a feature specific to AloeStackView either. Rather it’s something that applies to any UIScrollView, including UITableView.

I’m hesitant to add features such as this to AloeStackView so that the code footprint of the library can remain small. Small code size is an important feature of AloeStackView, so it needs to be carefully balanced against what features the library provides.

Having said that, many people may want to use pull-to-refresh with an AloeStackView. To address this, one approach might be to add a section to the documentation and perhaps also the example app that shows how to use a UIRefreshControl with AloeStackView.

This tutorial documentation would address the use case of this PR but without adding any additional weight and complexity to the library.

I would love to hear your thoughts on this. Also please let me know if you have any questions or concerns about anything I’ve said.

Thanks again for your contribution!

**Comment by marlimox** [comment]
@chuongtrh I’m going to close this PR. Please let me know if you’re interested in adding documentation to the project on how to use pull-to-refresh with AloeStackView.

**Comment by chuongtrh** [comment]
@marlimox Yes, pls adding documentation to the project on how to use pull-to-refresh with AloeStackView. Thanks.

**Pull request closed without merging**